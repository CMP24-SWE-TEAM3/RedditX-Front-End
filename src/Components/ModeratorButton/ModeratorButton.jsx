// imports
import React from 'react';
import {MdOutlineRemoveModerator} from 'react-icons/md';
import {ButtonModerator} from './ModeratorButton.styled';
import Dropdown from "react-bootstrap/Dropdown";
import {TbBoxModel2} from "react-icons/tb";
import OverlayTrigger from "react-bootstrap/OverlayTrigger";
import Tooltip from "react-bootstrap/Tooltip";
import {FiMail} from "react-icons/fi";

/**
 * Component to display the dropdown button with dropdown results
 *
 * @returns {React.Component}
 */
const ModeratorButton = () => {
    return (
        <><OverlayTrigger
            key={'bottom'}
            placement={'bottom'}
            overlay={
                <Tooltip id={`tooltip-bottom`}>Moderator</Tooltip>}>
            <ButtonModerator id={'moderator-button'}>
                <Dropdown.Toggle key={Math.random()} as={MdOutlineRemoveModerator}/>
                <Dropdown.Menu id={'moderator-menu'}>
                    <Dropdown.Item key={Math.random()} href="#"><span><TbBoxModel2/></span> <span>Mod Queue</span></Dropdown.Item>
                    <Dropdown.Item key={Math.random()} href="#"><span><FiMail/></span> <span>Modmail</span></Dropdown.Item>
                </Dropdown.Menu>
            </ButtonModerator>
        </OverlayTrigger>
        </>
    );
};

export default ModeratorButton;
