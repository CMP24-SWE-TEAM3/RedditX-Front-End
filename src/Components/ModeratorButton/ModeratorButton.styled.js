import styled from "styled-components";
import Dropdown from "react-bootstrap/Dropdown";

export const ButtonModerator = styled(Dropdown)`
  border: none;
  outline: none;
  //width: 38px;
  //height: 38px;
  margin: 0 8px;
  color:${({theme}) => theme.color.muted} !important;
  @media(max-width: 768px) {
    width: auto;
  }

  :hover {
    background: ${({theme}) => theme.background.post_background};
  }
  
  span {
    padding-right: 3px;
  }

  .dropdown-item {
    color: ${({theme}) => theme.color.primary} !important;
    :active {
      background: none !important;
      color: unset !important;
    }
    :hover{
      background: ${({theme}) => theme.background.post_background} !important;
    }
  }

  .dropdown-menu {
    background: ${({theme}) => theme.background.primary} !important;
  }

`