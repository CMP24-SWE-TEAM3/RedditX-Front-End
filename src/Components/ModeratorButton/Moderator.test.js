import {shallow} from 'enzyme';

// import Components
import ModeratorButton from "./ModeratorButton";

describe("Moderator button Component", () => {
    it("this is a test for Moderator Component", () => {
        expect(shallow(<ModeratorButton/>)).toMatchSnapshot();

    });
});