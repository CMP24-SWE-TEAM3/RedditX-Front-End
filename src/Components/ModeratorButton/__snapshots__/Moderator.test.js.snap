// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Moderator button Component this is a test for Moderator Component 1`] = `ShallowWrapper {}`;
